File: leetcode/cpp/ugly-number.cpp
// https://leetcode.com/problems/ugly-number/

class Solution {
public:
    bool isUgly(int num) {
        if (num < 1) {
            return false;
        }

        int set[] = {2, 3, 5},
            i = 0;

        for (i = 0; i < 3; ++i) {
            while (num % set[i] == 0) {
                num /= set[i];
            }
        }

        return (num == 1);
    }
};

File: leetcode/java/ugly-number.java
// https://leetcode.com/problems/ugly-number/

public class Solution {
    public boolean isUgly(int num) {
        if (num < 1) {
            return false;
        }

        int[] set = new int[] {2, 3, 5};
        for (int factor: set) {
            while (num % factor == 0) {
                num /= factor;
            }
        }

        return (num == 1);
    }
}
